packages/iTwinUI-react/src/core/Table/Table.tsx

```diff
@@ -752,7 +752,16 @@ export const Table = <
         })}
         {...ariaDataAttributes}
       >
-        <div className='iui-table-header-wrapper' ref={headerRef}>
+        <div
+          className='iui-table-header-wrapper'
+          ref={headerRef}
+          onScroll={() => {
+            if (headerRef.current && bodyRef.current) {
```

Comment:
This change was also needed for keyboard navigation.
Before:
![table-header-shift3-before](https://user-images.githubusercontent.com/36186912/182602955-17fabb15-a73c-4c82-8d37-e996112480ce.gif)

After:
![table-header-shift3](https://user-images.githubusercontent.com/36186912/182602973-709a8b9b-8c08-47de-b9c8-7858c6bad5d7.gif)